ThinkstCanary/thinkst_canary_modules/connector_thinkst_canary_alerts.py

```diff
@@ -0,0 +1,218 @@
+                    "timestamp": incident["updated_time"],
+                }
+            )
+
+            # get the events from the incident
+            events = description.pop("events", [])
+
+            # remove events_count and events_list from the description
+            description.pop("events_count")
+            description.pop("events_list")
+
+            for event in events:
+                record = {"incident_id": incident_id, "event_type": "event"}
+                record.update(description)
+                record.update(event)
+                records.append(record)
+
+        return records
+
+    def fetch_events(self) -> Generator[list, None, None]:
+        most_recent_id_seen = self.from_id
+        most_recent_timestamp_seen = None  # for the event lag calculation
+
+        try:
+            for page in self.fetch_pages(most_recent_id_seen):
+                events = self.extract_events(page)
+
+                last_event_id = page["max_updated_id"]
+                if most_recent_id_seen is None or last_event_id > most_recent_id_seen:
+                    most_recent_id_seen = last_event_id
+
+                if self.configuration.acknowledge:
+                    incidents_ids = [event["incident_id"] for event in events if event["event_type"] == "incident"]
+                    for incident_id in incidents_ids:
+                        self.acknowledge_incident(incident_id=incident_id)
+
+                if len(events) > 0:
+                    most_recent_timestamp_seen = dateutil.parser.parse(page["updated_std"])
+                    yield events
+
+        finally:
+            if self.from_id is None or most_recent_id_seen > self.from_id:
+                self.from_id = most_recent_id_seen
+
+                with self.context as cache:
+                    cache["most_recent_id_seen"] = most_recent_id_seen
+
+        if most_recent_timestamp_seen:
+            # no events - no lag
+            now = datetime.now(timezone.utc)
+            current_lag = now - most_recent_timestamp_seen
+            EVENTS_LAG.labels(intake_key=self.configuration.intake_key).set(int(current_lag.total_seconds()))
+
+    def next_batch(self) -> None:
+        # save the starting time
+        batch_start_time = time.time()
+
+        for events in self.fetch_events():
+            batch_of_events = [orjson.dumps(event).decode("utf-8") for event in events]
+
+            if len(batch_of_events) > 0:
+                self.log(
+                    message=f"Forwarded {len(batch_of_events)} events to the intake",
+                    level="info",
+                )
+                OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(batch_of_events))
+                self.push_events_to_intakes(events=batch_of_events)
+
+            else:
+                self.log(
+                    message="No events to forward",
+                    level="info",
+                )
```

Comment:
If no events were forwarded, we must reset the event lag (to avoid having a constant event lag while we are up-to-date)
```suggestion
                )
                EVENTS_LAG.labels(intake_key=self.configuration.intake_key).set(0)
```